src/smallestPrimeFactor.js

```diff
@@ -11,15 +11,48 @@
  * @param {number} num - A number
  * @returns {number} The smallest prime factor of num
  */
-function smallestPrimeFactor(num) {
+
+  function isPrime(num) {
+    if ( num === 1){
+       return false;}
+
+       if ( num === 2){return true;}
+
+     for ( let n = 2; n < num; n++){
+     if (num % n === 0){
+      return false;
+     }
+     return true;
+   }
+  }
+
+  function smallestPrimeFactors(num) {
```

Comment:
You don't need to calculate _every_ prime factor to find the _smallest_ prime factor. How would you find the smallest prime factor using pen and paper?

In fact, if you think about how you determine whether a particular number is prime using pen and paper, you're more likely to implement `isPrime` in terms of `smallestPrimeFactor` rather than the other way around (like you've done here). That is, if you pay attention to how you as a human determine whether a number is prime, it's much more like you're doing the following:

```javascript
function isPrime(num) {
  if (num < 2) {
    return false;
  }

  return smallestPrimeFactor(num) === num;
}
```

Reply:
(Your `isPrime` function already finds the smallest prime factor.)